import/index_java/src/org/vufind/index/CallNumberTools.java

```diff
@@ -325,7 +325,7 @@ public Set<String> getDeweyNumber(Record record, String fieldSpec, String precis
                 float currentVal = Float.parseFloat(callNum.getClassification());
 
                 // Round the call number value to the specified precision:
-                Float finalVal = new Float(Math.floor(currentVal / precision) * precision);
+                Float finalVal = Double.valueOf(Math.floor(currentVal / precision) * precision).floatValue();
```

Comment:
Math.floor returns a double, so this appears to be an appropriate way to convert it to float.